### ISSUE TITLE
Remove terraform.local from documentation and examples

### ISSUE THREAD

**Issue by lekaf974** [wundergraph/terraform-provider-cosmo#6]
Following the tutorial `examples/provider` (release v0.0.2)

The terraform init command is not working properly

```bash
[provider]$ terraform init
Initializing the backend...
Initializing modules...
- cosmo-federated-graph in ../../modules/cosmo-federated-graph
- data_cosmo_federated_graph in ../data-sources/cosmo_federated_graph
- resource_cosmo_federated_graph in ../resources/cosmo_federated_graph
- resource_cosmo_namespace in ../resources/cosmo_namespace
- resource_cosmo_subgraph in ../resources/cosmo_subgraph
Initializing provider plugins...
- Reusing previous version of wundergraph/cosmo from the dependency lock file
- Finding terraform.local/wundergraph/cosmo versions matching "0.0.1"...
- Reusing previous version of hashicorp/random from the dependency lock file
- Installing wundergraph/cosmo v0.0.1...
- Installed wundergraph/cosmo v0.0.1 (self-signed, key ID 5F7E617DB4FE69A3)
- Installing hashicorp/random v3.6.3...
- Installed hashicorp/random v3.6.3 (signed by HashiCorp)
Partner and community providers are signed by their developers.
If you'd like to know more about provider signing, you can read about it here:
https://www.terraform.io/docs/cli/plugins/signing.html
╷
│ Error: Failed to query available provider packages
│ 
│ Could not retrieve the list of available versions for provider terraform.local/wundergraph/cosmo: could not connect to terraform.local: failed to request discovery document: Get "https://terraform.local/.well-known/terraform.json": dial tcp: lookup terraform.local on 192.168.3.1:53: no such host

```

**Comment by lekaf974** [comment]
@AndreasZeissner created this issue and send a PR if it make sense

**Comment by AndreasZeissner** [comment]
Hi @lekaf974,

thanks for pointing that out. The examples are meant to be used with a local build of the terraform provider which you can do by running `make clean build install` and then e.g. `make e2e`. 

It is a bit inconvenient at the moment, I agree. Please use the `USE PROVIDER` button here https://registry.terraform.io/providers/wundergraph/cosmo/latest to get the latest published version.

```
terraform {
  required_providers {
    cosmo = {
      source = "wundergraph/cosmo"
      version = "0.0.2"
    }
  }
}

provider "cosmo" {
  # Configuration options
}
```

You might open a PR for sure, I would just want these workflows to still exist as it's pretty handy during development. Apart from that I would have a look into making it more convenient in the future.

**Comment by mevrin-ueat** [comment]
the only think to do is to ensure that `terraform.local` is just removed from examples and documentation when it is released on terraform  right ?

for example https://registry.terraform.io/providers/wundergraph/cosmo/latest/docs 

![image](https://github.com/user-attachments/assets/b7124eb1-f2ed-467f-8ee8-5e71c2a7495e)

**Comment by AndreasZeissner** [comment]
Hi,

@mevrin-ueat yes indeed. But at time of generating it's not yet clear which version the next one is. The tagging happens during release. The `terraform.local` is only used to reference the local plugin store.  

have a look here:

- https://github.com/wundergraph/terraform-provider-cosmo/blob/andi/eng-5704-inquiring-about-resources-for-contracts-in-terraform/docs/resources/federated_graph.md
- https://github.com/wundergraph/terraform-provider-cosmo/pull/7

I moved the provider into an extra file which will avoid that it's rendered and only the resource is present. I would say this does the job and keeps maintenance and confusion low.

**Comment by lekaf974** [comment]
Closing PR created

**Comment by AndreasZeissner** [comment]
@lekaf974  @mevrin-ueat,

I recently released version 0.1.0: 

- https://registry.terraform.io/providers/wundergraph/cosmo/latest/docs
- https://github.com/wundergraph/terraform-provider-cosmo/releases/tag/v0.1.0

Things are way cleaner now.
